src/main/java/carpet/helpers/OptimizedExplosion.java

```diff
@@ -457,38 +458,44 @@ private static boolean checkAffectedPosition(Explosion e, double xRel, double yR
         {
             posMutable.set(posX, posY, posZ);
 
-            // Don't query already cached positions again from the world
-            BlockState state = stateCache.get(posMutable);
-            FluidState fluid = fluidCache.get(posMutable);
             BlockPos posImmutable = null;
 
-            if (state == null)
+            // Don't query already cached positions again from the world
+            if (!blastResCache.containsKey(posMutable))
             {
                 posImmutable = posMutable.toImmutable();
-                state = eAccess.getWorld().getBlockState(posImmutable);
-                stateCache.put(posImmutable, state);
-                fluid = eAccess.getWorld().getFluidState(posImmutable);
-                fluidCache.put(posImmutable, fluid);
-            }
-
-            if (state.getMaterial() != Material.AIR)
-            {
-                float resistance = Math.max(state.getBlock().getBlastResistance(), fluid.getBlastResistance());
-
-                if (eAccess.getEntity() != null)
+                BlockState state = world.getBlockState(posImmutable);
+                FluidState fluid = world.getFluidState(posImmutable);
+                Float resistance = null;
+                if (state.getMaterial() != Material.AIR)
```

Comment:
I believe there is world.isAir() function - not sure which would be faster here

Reply:
Looking at that particular code again, I think I can save some more calculations by moving getFluidState into that if statement. I will make that change and then test world.isAir() using Spark to see which is faster.